discovery-infra/utils.py

```diff
@@ -368,3 +366,31 @@ def wrapped(*args, **kwargs):
                 raise
         return wrapped
     return decorator
+
+
+@contextmanager
+def file_lock_context(filepath='/tmp/discovery-infra.lock', timeout=300):
+    logging.getLogger('filelock').setLevel(logging.INFO)
+
+    lock = filelock.FileLock(filepath, timeout)
+    try:
+        lock.acquire()
+    except filelock.Timeout:
```

Comment:
can u explain it? if somebody took lock we will delete it?

Reply:
if acquiring the lock is passed a certain timeout that we know it is not reasonable to wait for that long, we delete the lock file and acquire it. this for cases user aborts CI while lock is acquired. in such cases the lock is not release and we need to override it